docs/SemanticARC.md

```diff
@@ -128,8 +145,41 @@ In order to simplify this, we will make the following changes:
    * @unowned @safe
    * @forwarding
 
+[EE] We really need a good definition of all these conventions here.
+I strongly believe that the local/global notation would help a lot for a good formal definition of the conventions.
+It's hard for the compiler to reason anything about global RCs. But local RCs are easily statically computable and for me, semantic ARC is all about dealing with local RCs.
+
+So we could define:
+
+@owned operand: local RC must be 1 and is decremented to 0 by the instruction, potentially destroying the object.
+@owned result: Returns the object with a local RC of 1
+@guaranteed operand: local RC must be 1 and is not modified by the instruction
+@unowned operand: local RC must be 0 and is not modified by the instruction.
+@unowned result: Returns the object with a local RC of 0
+@forwarding: Does not change the local RC which can be either 0 or 1
+
+Note that the local RC can only be 0 or 1. It doesn't make sense to have a local RC > 1.
+
+ad @unowned: what do you mean with @unsafe and @safe
```

Comment:
I am not 100% sure how you can produce it with swift, but it would look like:

sil @unowned_user : $@convention(thin) (Builtin.NativeObject) -> ()

IIRC, Builtin.NativeObject here is unowned unsafe. This is the equivalent from objective c of __unsafe_unretained. I do not think that it is inherently obvious that Builtin.NativeObject here has that semantics. On the other hand if I have:

sil @unowned_user : $@convention(thin) (@unowned_unsafe Builtin.NativeObject) -> ()

It is really clear that there is something weird going on here and that one should look up the convention.